canopy_store.ml

```diff
@@ -102,8 +111,7 @@ module Store (CTX: Irmin_mirage.CONTEXT) (INFL: Git.Inflate.S) = struct
         match C.of_string ~base_uuid ~uri ~content ~created ~updated with
         | C.Ok article -> KeyMap.add key (`Article article) cache
         | C.Error error ->
-          Log.warn (fun f -> f "Error while parsing %s: %s" uri error) ;
-          cache
+          check_if_redirect content cache key (fun f -> f "Error while parsing %s: %s" uri error)
```

Comment:
I do like this PR -- but can we have `check_redirect` be `string -> string option` (as in, pass only `content` and return either `None` or `Some path`, and do the `add` or `warn` still in here?